### PR TITLE
Improved static code analysis

### PR DESCRIPTION
**Pull request by chives**

### CODE REVIEW

**Comment by chives** [comment]
@fsi-open

File: Form/AbstractTranslatableExtension.php [hunk 1]
```diff
@@ -42,8 +42,10 @@ public function finishView(FormView $view, FormInterface $form, array $options)
         }
 
         $view->vars['translatable'] = true;
-        if (!$this->hasCurrentValue($view, $form, $options) ||
-            $this->translatableFormHelper->isFormDataInCurrentLocale($parent)) {
+        if (
```
Comment (szymach) [inline Form/AbstractTranslatableExtension.php:45]:
```php
        if (
            false === $this->hasCurrentValue($view, $form, $options)
           || true === $this->translatableFormHelper->isFormDataInCurrentLocale($parent)
        ) {
```

**Comment by chives** [comment]
@fsi-open

File: Manager/LocaleManager.php [hunk 2]
```diff
@@ -66,13 +75,15 @@ public function getLocale(): ?string
...
     {
-        $eventManager = $this->managerRegistry->getManager()->getEventManager();
+        $objectManager = $this->managerRegistry->getManager();
+        if (false === $objectManager instanceof EntityManagerInterface) {
+            throw new RuntimeException(
+                sprintf("Expected %s but got %s", EntityManagerInterface::class, get_class($objectManager))
+            );
+        }
+        $eventManager = $objectManager->getEventManager();
         foreach ($eventManager->getListeners() as $listeners) {
```
Comment (szymach) [inline Manager/LocaleManager.php:87]: Yeah much more verbatim, but also clearer:
```php
$listener = array_reduce(
    $eventManager->getListeners(),
    function (?TranslatableListener $accumulator, array $listeners): ?TranslatableListener {
        if (null !== $accumulator) {
            return $accumulator;
        }

        return $this->findTranslatableListener($listeners);
    }
);

if (false === $listener instanceof TranslatableListener) {
    throw new RuntimeException(
        'Translatable extension is not enabled in "fsi_doctrine_extensions" section of "config.yml"'
    );
}

return $listener;

private function findTranslatableListener(array $listeners): ?TranslatableListener
{
    return array_reduce(
        $listeners,
        function (?TranslatableListener $accumulator, $listener): ?TranslatableListener {
            if (null !== $accumulator) {
                return $accumulator;
            }

            if (true === $listener instanceof TranslatableListener) {
                $accumulator = $listener;
            }

            return $accumulator;
        }
    );
}
```

File: Form/TranslatableFormHelper.php [hunk 3]
```diff
@@ -134,23 +138,38 @@ private function hasFormTranslatableProperty(
...
     {
-        return $this->managerRegistry->getManagerForClass($class);
+        $objectManager = $this->managerRegistry->getManagerForClass($class);
+        if (null === $objectManager) {
+            return null;
+        }
+
+        if (false === $objectManager instanceof EntityManagerInterface) {
+            throw new RuntimeException(
+                sprintf("Expected %s but got %s", EntityManagerInterface::class, get_class($objectManager))
```
[the remark this answers is not in the document]
Comment (chives) [reply Form/TranslatableFormHelper.php:157]: it depends if the line fits in 120 chars limit

**Comment by chives** [comment]
@fsi-open

**Review by szymach — APPROVED** [review]

**Review by rn0 — APPROVED** [review]

**Pull request merged**